src/cabal-details.js

```diff
@@ -662,6 +732,28 @@ class CabalDetails extends EventEmitter {
       })
     })
 
+    // notify when a user has archived a channel
+    this.registerListener(cabal.archives.events, 'archive', (channel, reason, key) => {
+      const user = this.users[key]
+      if (key !== this.user.key && (!user || (!user.isModerator() && !user.isAdmin()))) { return }
+      if (!this.channels[channel]) {
+        this.channels[channel] = new ChannelDetails(this.core, channel)
+      }
+      this.channels[channel].archive()
+      this._emitUpdate('channel-archive', { channel, reason, key, isLocal: key === this.user.key })
+    })
+
+    // notify when a user has restored an archived channel
+    this.registerListener(cabal.archives.events, 'unarchive', (channel, reason, key) => {
+      const user = this.users[key]
+      if (key !== this.user.key && (!user || (!user.isModerator() && !user.isAdmin()))) { return }
```

Comment:
would it be useful to have a method on `user` that handles this `user.isModerator() || user.isAdmin()` check?

Reply:
done! added `user.canModerate([channel])`